lib/hacluster.pm

```diff
@@ -1248,4 +1250,95 @@ sub cluster_status_matches_regex {
     }
 }
 
+=head3 crm_maintenance_status
+
+    crm_maintenance_status();
+
+Check maintenance mode status. Returns true (maintenance active) or false (maintenance inactive).
+Croaks if unknown status is received.
+
+=over 1
+
+=back
+=cut
+
+sub crm_maintenance_status {
+    my $cmd_output = script_output('crm configure show cib-bootstrap-options | grep maintenance-mode');
+    $cmd_output =~ s/\s//g;
+    my $status = (split('=', $cmd_output))[-1];
+    croak "CRM returned unrecognized status: '$status'" unless grep(/^$status$/, ('false', 'true'));
+    return $status;
+}
+
+=head3 crm_wait_for_maintenance
+
+    crm_wait_for_maintenance(target_state=>$target_state, [loop_sleep=>$loop_sleep, timeout=>$timeout]);
+
+Wait for maintenance to be turned on or off. Croaks on timeout.
+
+B<target_state> Target state of the maintenance mode (true/false)
+B<loop_sleep> Override default sleep value between checks
+B<timeout> Override default timeout value
+
+=over 1
+
+=back
+=cut
+
+sub crm_wait_for_maintenance {
+    my (%args) = @_;
+    my $timeout = $args{timeout} // bmwqemu::scale_timeout(30);
+    my $target_state = $args{'target_state'};
```

Comment:
There's maybe no need to create a variable like `$target_state` but we can directly use everywhere directly the `$args{'target_state'};`

Reply:
Yes I know. I just prefer to have definitions at the top so the code is self explanatory what are available arguments without reading the function doc.

I'll remove it.